judge/views/problem.py

```diff
@@ -422,7 +422,7 @@ def get_context_data(self, **kwargs):
         else:
             context['hot_problems'] = None
             context['point_start'], context['point_end'], context['point_values'] = 0, 0, {}
-            context['hide_contest_scoreboard'] = self.contest.hide_scoreboard
+            context['hide_contest_scoreboard'] = self.contest.scoreboard_visibility == "Contest"
```

Comment:
This leaks the problem's user count if the visibility is set to `Participation`: https://github.com/DMOJ/online-judge/blob/master/templates/problem/list.html#L292